pvlib/spectrum/mismatch.py

```diff
@@ -235,3 +235,160 @@ def integrate(e):
+                'poa_sky_diffuse': {'c': c2, 'a': a2, 'b': b2},
+                'poa_ground_diffuse': {'c': c3, 'a': a3, 'b': b3}
+            }
+            # Using a pd.DataFrame
+            model_parameters = pd.DataFrame({
+                'poa_direct': [c1, a1, b1],
+                'poa_sky_diffuse':  [c2, a2, b2],
+                'poa_ground_diffuse': [c3, a3, b3]},
+                index=('c', 'a', 'b'))
+
+        ``c``, ``a`` and ``b`` must be scalar.
+
+        Unspecified parameters for an irradiance component (`'poa_direct'`,
+        `'poa_sky_diffuse'`, or `'poa_ground_diffuse'`) will cause ``np.nan``
+        to be returned in the corresponding result.
+
+    Returns
+    -------
+    Modifiers : pd.DataFrame (iterable input) or dict (scalar input) of numeric
+        Mismatch modifiers for direct, sky diffuse and ground diffuse
+        irradiances, with indexes `'poa_direct'`, `'poa_sky_diffuse'`,
+        `'poa_ground_diffuse'`.
+        Each mismatch modifier should be multiplied by its corresponding
+        POA component.
+
+    Raises
+    ------
+    ValueError
+        If ``model_parameters`` is not suitable. See examples given above.
+    TypeError
+        If neither ``module_type`` nor ``model_parameters`` are given.
+    TypeError
+        If both ``module_type`` and ``model_parameters`` are provided.
+    NotImplementedError
+        If ``module_type`` is not found in internal table of parameters.
+
+    Notes
+    -----
+    The mismatch modifier is defined as
+
+    .. math:: M = c \cdot \exp( a \cdot (K_t - 0.74) + b \cdot (AM - 1.5) )
+
+    where :math:`c`, :math:`a` and :math:`b` are the model parameters,
+    different for each irradiance component.
+
+    References
+    ----------
+    .. [1] Martín, N. and Ruiz, J.M. (1999), A new method for the spectral
+       characterisation of PV modules. Prog. Photovolt: Res. Appl., 7: 299-310.
+       :doi:`10.1002/(SICI)1099-159X(199907/08)7:4<299::AID-PIP260>3.0.CO;2-0`
+
+    See Also
+    --------
+    pvlib.irradiance.clearness_index
+    pvlib.atmosphere.get_relative_airmass
+    pvlib.atmosphere.get_absolute_airmass
+    pvlib.atmosphere.first_solar
+    """
+    # Note tests for this function are prefixed with test_martin_ruiz_mm_*
+
+    IRRAD_COMPONENTS = ('poa_direct', 'poa_sky_diffuse', 'poa_ground_diffuse')
+    # Fitting parameters directly from [1]_
+    MARTIN_RUIZ_PARAMS = pd.DataFrame(
+        index=('monosi', 'polysi', 'asi'),
+        columns=pd.MultiIndex.from_product([IRRAD_COMPONENTS,
+                                           ('c', 'a', 'b')]),
+        data=[  # Direct(c,a,b)  | Sky diffuse(c,a,b) | Ground diffuse(c,a,b)
+            [1.029, -.313, 524e-5, .764, -.882, -.0204, .970, -.244, .0129],
+            [1.029, -.311, 626e-5, .764, -.929, -.0192, .970, -.270, .0158],
+            [1.024, -.222, 920e-5, .840, -.728, -.0183, .989, -.219, .0179],
+        ])
+
+    # Argument validation and choose components and model parameters
```

Comment:
I have tried what you say, but I don't get nice results - when both are provided, one of the inputs is ignored without telling so; and when neither of them are, `_params is not defined` is raised, which could be a good error message.
In my humble opinion, being verbatim in the interface is not a problem, and is better than seeing a variable you did not write is not being defined. It will save time for the user, essentially.

Reply:
numpy and scipy use the None, None default pattern, where if both are None the function fails, so I think we're among good company.